test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -68,6 +68,25 @@ object TestAspect {
       }
     }
 
+  /**
+   * An aspect that repeats the test a specified number of times, ensuring it
+   * is stable ("non-flaky").
+   */
+  def nonFlaky(n0: Int): TestAspectPoly =
+    new TestAspectPoly {
+      def apply[R >: Nothing <: Any, E >: Nothing <: Any](test: ZIO[R, E, TestResult]): ZIO[R, E, TestResult] = {
+        def repeat(n: Int): ZIO[R, E, TestResult] =
```

Comment:
You should be able to put a `@tailrec` here, no?

Reply:
It's monadically tail recursive, so the compiler can't accept that as `@tailrec`.

Reply:
Ah, that's true...